tests/foreman/api/test_ansible.py

```diff
@@ -123,3 +129,56 @@ def test_positive_ansible_job_on_host(target_sat, module_org, rhel_contenthost):
     )
     result = target_sat.api.JobInvocation(id=job['id']).read()
     assert result.succeeded == 1
+
+    target_sat.api.Host(id=target_host.id).remove_ansible_role(data={'ansible_role_id': role_id})
+    host_roles = target_host.list_ansible_roles()
+    assert len(host_roles) == 0
+
+
+@pytest.mark.e2e
+@pytest.mark.stream
+@pytest.mark.tier2
+def test_add_and_remove_ansible_role_hostgroup(target_sat):
+    """
+    Test add and remove functionality for ansible roles in hostgroup via API
+
+    :id: 7672cf86-fa31-11ed-855a-0fd307d2d66b
+
+    :Steps:
+        1. Create a hostgroup
+        2. Sync few ansible roles
+        3. Assign a few ansible roles with the host group
+        4. Add some ansible role with the host group
+        5. Remove the added ansible roles from the host group
+
+    :expectedresults:
+        1. Ansible role assign/add/remove functionality should work as expected in API
+
+    :BZ: 2164400
+    """
+    ROLE_NAMES = [
+        'theforeman.foreman_scap_client',
+        'redhat.satellite.hostgroups',
+        'RedHatInsights.insights-client',
+    ]
+    ROLES = []
+    hg_name = gen_string('alpha')
+    hg = target_sat.api.HostGroup(name=hg_name).create()
+    assert hg.name == hg_name
+    proxy_id = target_sat.nailgun_smart_proxy.id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': proxy_id, 'role_names': ROLE_NAMES})
+    for role in ROLE_NAMES:
+        role_id = target_sat.api.AnsibleRoles().search(query={'search': f'name={role}'})[0].id
+        ROLES.append(role_id)
+
+    target_sat.api.HostGroup(id=hg.id).assign_ansible_roles(data={'ansible_role_ids': ROLES[:2]})
+    for a, b in zip(target_sat.api.HostGroup(id=hg.id).list_ansible_roles(), ROLE_NAMES[:2]):
```

Comment:
Nice use of Zip. But provided some meaningful names instead of  a/b.

Reply:
@jyejare Yes, I knew it would be raised in review, I couldn't find any related names as we're validating roles names from API with defined constant, so wasn't getting any suitable names here, and it will be extend the line unnecessary that's why I preferred a/b, do you've any suggestions for the names? :)

Reply:
a can be `role` and `b` can be `role_name`.

Reply:
Actually, both returns role names only, this names would be little confusing later, so I kept this iterative vars as r1 and r2 instead of a/b, I wouldn't want extend lines here so would like to keep it short, let me know if you still find this vars confusing

Reply:
Actually, both returns role names, so it'll be little confusing if we assign this names, instead I'd prefer something short like r1 and r2 instead for iterative variables, it keeps this loop in single line, let me know if you think it won't make sense?

Reply:
ok, lets not waste more time on it :)